src/core/reaction_ensemble.cpp

```diff
@@ -0,0 +1,2061 @@
+//method according to smith94x for the reaction ensemble at constant volume and temperature, for the reaction ensemble at constant pressure additionally employ a barostat!
+//NOTE: a chemical reaction consists of a forward and backward reaction. Here both reactions have to be defined seperately.
+//The extent of the reaction is here chosen to be +1.
+//If the reaction trial move for a dissociation of HA is accepted then there is one more dissociated ion pair H+ and A-
+//NOTE: generic_oneway_reaction does not break bonds for simple reactions. as long as there are no reactions like 2A -->B where one of the reacting A particles occurs in the polymer (think of bond breakages if the monomer in the polymer gets deleted in the reaction). This constraint is not of fundamental reason, but there would be a need for a rule for such "collision" reactions (a reaction like the one above).
+//NOTE: paricle types have to start at one and have to increase by one for every type otherwise the function hide_particle cannot work correctly. Through adding 100 in the function hide_particle we ensure that the function works correctly if the particle types are monotonically increasing and if the largest particle type is smaller than 100.
+
+#include "reaction_ensemble.hpp"
+#include "random.hpp" //for random numbers
+#include "energy.hpp"	//for energies
+#include "external_potential.hpp" //for energies
+#include "global.hpp" //for access to global variables
+#include "particle_data.hpp" //for particle creation, modification
+#include "statistics.hpp" //for distto
+#include "integrate.hpp" //for time_step
+#include <stdlib.h>  // for qsort()
+#include <stdio.h> //for getline()
+#include <iostream> //for std::cout
+#include <fstream> //for std::ifstream, std::ofstream for input output into files
+#include "utils.hpp" // for PI
+
+reaction_system current_reaction_system={.nr_single_reactions=0, .reactions=NULL , .type_index=NULL, .nr_different_types=0, .charges_of_types=NULL, .standard_pressure_in_simulation_units=-10, .temperature_reaction_ensemble=-10.0, .exclusion_radius=0.0, .volume=-10, .box_is_cylindric_around_z_axis=false, .cyl_radius=-10, .cyl_x=-10,. cyl_y=-10, .box_has_wall_constraints=false, .slab_start_z=-10, .slab_end_z=-10}; //the standard_pressure_in_simulation_units is an input parameter for the reaction ensemble
+
+//global variable
+bool system_is_in_1_over_t_regime=false;
+
+int number_of_times_in_the_same_bin=0;//for avoiding becoming trapped in HMC-Wang-Landau
+int last_bin_index=-10;//for avoiding becoming trapped in HMC-Wang-Landau
+int invalid_charge=-10000;
+
+double minimum_average_of_hisogram_before_removal_of_unused_bins=1000.0; //make this accessible from interface (use 90 for free polymer, 1000 for rod system, this is determined by trial and error)
+
```

Comment:
Yes as I noted in the comment this should be accessible from the interface. I would rather drop the support for simulating polymers with Wang-Landau energy reweighting simulations for now, since this not trivial to test for correctness.